app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug6732_Spec.ts

```
@@ -0,0 +1,21 @@
+import * as _ from "../../../../support/Objects/ObjectsCore";
+
+const { agHelper, apiPage } = _;
+
+describe("this.params in IIFE function in API editor", () => {
+  it("1. this.params should be available in IIFE function in API editor", () => {
+    apiPage.CreateApi("Api1", "GET");
+
+    apiPage.SelectPaneTab("Params");
+    apiPage.EnterParams(
+      "page",
+      "{{(function () { return this.params.key; })()}}",
+      0,
+      false,
+    );
+
+    // the value is evaluated correctly only during execution
+    // in this case, we just want to ensure that this.params is an object and present
+    agHelper.VerifyEvaluatedValue("undefined", false);
```

Showing undefined is the bug too. I dint quite get the fix here.

---

The issue was `this.params` was being evaluated as `undefined` and hence accessing `this.params.something` raised an error `Cannot read properties of undefined`. The fix makes sure that `this.params` is a an empty object to suppress this error. We can't be sure of the exact value of `this.params` here on API editor page as `this.params` is contextual to the invocation, where the API is being called passing the params object.

In the test, we're making sure that the value of `this.params.key` gets evaluated without any errors, and the value will be undefined because `this.params` is an empty object here on this page.

---

Ok, why cant we send some value to this API param & see if its evaluated - instead of undefined?

---

Before fix also it was undefined & after fix also its undefined - makes it confusing.